ic-agent/src/identity/mod.rs

```diff
@@ -24,19 +26,75 @@ pub struct Signature {
     pub signature: Option<Vec<u8>>,
 }
 
-/// An Identity takes a request id and returns the [Signature]. It knows or
-/// represents the Principal of the sender.
+/// An `Identity` produces [`Signatures`](Signature) for requests or delegations. It knows or
+/// represents the [`Principal`] of the sender.
 ///
-/// Agents are assigned a single Identity object, but there can be multiple
+/// [`Agents`](crate::Agent) are assigned a single `Identity` object, but there can be multiple
 /// identities used.
 pub trait Identity: Send + Sync {
     /// Returns a sender, ie. the Principal ID that is used to sign a request.
     ///
     /// Only one sender can be used per request.
     fn sender(&self) -> Result<Principal, String>;
 
+    /// Produce the public key commonly returned in [`Signature`].
+    ///
+    /// May return `None` when [`sign`](Identity::sign) would return `Some`.
+    fn public_key(&self) -> Option<Vec<u8>>;
+
     /// Sign a request ID derived from a content map.
     ///
     /// Implementors should call `content.to_request_id().signable()` for the actual bytes that need to be signed.
     fn sign(&self, content: &EnvelopeContent) -> Result<Signature, String>;
+
+    /// Sign a delegation to let another key be used to authenticate [`sender`](Identity::sender).
+    ///
+    /// Not all `Identity` implementations support this operation.
+    ///
+    /// Implementors should call `content.signable()` for the actual bytes that need to be signed.
+    fn sign_delegation(&self, content: &Delegation) -> Result<Signature, String> {
+        let _ = content; // silence unused warning
+        Err(String::from("unsupported"))
+    }
+
+    /// Sign arbitrary bytes.
+    ///
+    /// Not all `Identity` implementations support this operation.
```

Comment:
Please document that all identities, except the anonymous identity, implemented in `ic-agent` supports this method.